roles/gitea/tasks/install.yml

```diff
@@ -0,0 +1,166 @@
+# Prepare to install Gitea: create user and directory structure
+
+- name: Create user gitea
+  user:
+    name: gitea
+    comment: Gitea daemon account
+    groups: gitea
+    home: "{{ gitea_root_directory }}"
+  tags:
+    - pre-install
+
+- name: Create Gitea directory structure
+  file:
+    path: "{{ gitea_root_directory }}/{{item.item}}"
+    state: directory
+    owner: gitea
+    group: gitea
+  with_items: "{{ gitea_subdirectories }}"
+  tags:
+    - pre-install
+
+- name: Make directories data, indexers, and log writable
+  file:
+    path: "{{ gitea_root_directory }}/{{item.item}}"
+    mode: 0750
+  with_items:
+    - data
+    - indexers
+    - log
+  tags:
+    - pre-install
+
+# Download, verify, and link Gitea binary
+
+- name: Fail if we detect unknown architecture
+  fail:
+    msg: "Could not find a binary for the CPU architecture \"{{ ansible_architecture }}\""
+  when: gitea_iset_suffix == "unknown"
+
+- name: Download Gitea binary
+  get_url:
+    url: "{{ gitea_download_url }}"
+    dest: "{{ gitea_install_path }}"
+    mode: 0775
+  tags:
+    - install
+  when: internet_available
+
+- name: Download Gitea GPG signature
+  get_url:
+    url: "{{ gitea_integrity_url }}"
+    dest: "{{ gitea_checksum_path }}"
+  tags:
+    - install
+  when: internet_available
+
+- name: Verify Gitea binary with GPG signature
+  shell: |
+    gpg --keyserver pgp.mit.edu --recv 7C9E68152594688862D62AF62D9AE806EC1592E2
```

Comment:
does the hash here belong in `/defaults/main.yml`?

Reply:
This is Gitea's GPG key; I got it from [here](https://docs.gitea.io/en-us/install-from-binary/). I believe it works for all binaries.